kernel/src/memop.rs

```diff
@@ -37,6 +37,54 @@ pub fn memop(process: &mut Process) -> ReturnCode {
         /// Op Type 6: Grant region begin
         6 => ReturnCode::SuccessWithValue { value: process.kernel_memory_break() as usize },
 
+        /// Op Type 7: Number of defined writeable regions in the TBF header.
+        7 => ReturnCode::SuccessWithValue { value: process.number_writeable_flash_regions() },
+
+        /// Op Type 8: The start address of the writeable region indexed by r1.
+        8 => {
+            let flash_start = process.flash_start() as usize;
+            let (offset, size) = process.get_writeable_flash_region(r1);
+            if size == 0 {
+                // Returning NULL is actually an error. If we return a different
+                // return code, then we can't tell between an error and just
+                // a pointer very high in the address space.
+                ReturnCode::SUCCESS
```

Comment:
I don't love that, but I do agree that this is confusing as written, particularly because of line 62, which is doing the same thing only without this comment. Perhaps a reasonable compromise would be something like:

```rust
let nullptr_as_failure = 0;
ReturnCode::SuccessWithValue { value: nullptr_as_failure }
```

which gives someone reading this code more clarity that this isn't actually successful